solr/core/src/java/org/apache/solr/core/QuerySenderListener.java

```diff
@@ -102,4 +103,48 @@ public void close() {}
     }
     log.info("QuerySenderListener done.");
   }
+
+  private ArrayList<NamedList<Object>> queries2list(ArrayList<Object> queries) {
```

Comment:
I'm somewhat confused about the general structure of this method. Trying to read what's going on before vs after, it seems like the behavior for json would remain unchanged but we're fixing a bug specific to xml?

Reply:
It's the other way round - the bug was with JSON config received via the Config API. XML from `solrconfig.xml` works as before, and there's now support for JSON.